assets/js/theme/account.js

```diff
@@ -201,6 +208,128 @@ export default class Account extends PageManager {
+                paymentMethodValidator.remove($stateElement);
+            }
+
+            if ($last) {
+                paymentMethodValidator.remove($last);
+            }
+
+            if ($field.is('select')) {
+                $last = field;
+                Validators.setStateCountryValidation(paymentMethodValidator, field);
+            } else {
+                Validators.cleanUpStateValidation(field);
+            }
+        });
+
+        // Use credit card number input listener to highlight credit card type
+        $(`${paymentMethodSelector} input[name="credit_card_number"]`).on('keyup', ({ target }) => {
+            const cardType = creditCardType(target.value);
+            if (cardType) {
+                $(`${paymentMethodSelector} img[alt="${cardType}"`).siblings().css('opacity', '.2');
+            } else {
+                $(`${paymentMethodSelector} img`).css('opacity', '1');
+            }
+        });
+
+        // Set of credit card validation
+        CCValidators.setCreditCardNumberValidation(paymentMethodValidator, `${paymentMethodSelector} input[name="credit_card_number"]`, this.context.creditCardNumber);
+        CCValidators.setExpirationValidation(paymentMethodValidator, `${paymentMethodSelector} input[name="expiration"]`, this.context.expiration);
+        CCValidators.setNameOnCardValidation(paymentMethodValidator, `${paymentMethodSelector} input[name="name_on_card"]`, this.context.nameOnCard);
+        CCValidators.setCvvValidation(paymentMethodValidator, `${paymentMethodSelector} input[name="cvv"]`, this.context.cvv);
+
+        // Set of credit card format
+        CCFormatters.setCreditCardNumberFormat(`${paymentMethodSelector} input[name="credit_card_number"]`);
+        CCFormatters.setExpirationFormat(`${paymentMethodSelector} input[name="expiration"`);
+
+        // Billing address validation
+        paymentMethodValidator.add(validationModel);
+
+        $paymentMethodForm.on('submit', event => {
+            event.preventDefault();
+            // Perform final form validation
+            paymentMethodValidator.performCheck();
+            if (paymentMethodValidator.areAll('valid')) {
+                let country;
+                let state;
+                // Serialize form data and reduce it to object
+                const data = $paymentMethodForm
+                    .serializeArray()
+                    .reduce((obj, item) => {
+                        const refObj = obj;
+                        // Map dynamic fields to api payload
+                        switch (item.name) {
+                        case 'FormField[2][4]':
+                            refObj.first_name = item.value;
+                            break;
+                        case 'FormField[2][5]':
+                            refObj.last_name = item.value;
+                            break;
+                        case 'FormField[2][6]':
```

Comment:
I would really like to avoid doing this if we can help it. 
Is there a way that we can either: 
- Build the form manually 
- Get the backend to send proper `names` over the wire so we can avoid or simplify this mapping?

🍹

Reply:
I agree we need a better way of building these switch cases instead of relying on numeric keys.

Reply:
Hey @junedkazi @danieldelcore , the best alternative will be to create those input fields manually so I have a better control of the form names.